osquery/tables/system/linux/disk_encryption.cpp

```diff
@@ -0,0 +1,100 @@
+#include <unistd.h>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/sql.h>
+
+extern "C" {
+#include <libcryptsetup.h>
+
+crypt_status_info crypt_status(struct crypt_device *cd, const char *name);
+
+#ifdef CENTOS_CENTOS6
+int crypt_init_by_name(struct crypt_device **cd, const char *name);
+#define crypt_init(cd, name) crypt_init_by_name(&cd, name.c_str())
+#else
+int crypt_init_by_name_and_header(struct crypt_device **cd,
+				  const char *name,
+				  const char *header_device);
+#define crypt_init(cd, name) \
```

Comment:
Would it be easier to call the non-crypt_init versions inline along with inline `#ifdef`s?